### ISSUE TITLE
[ffigen] Regression: v14.0.0 does not generate bindings with a "plain" `include-directives`

### ISSUE THREAD

**Issue by Levi-Lesches** [dart-lang/native#1543]
@dcharkes I tried looking around some commits and the changelog, but couldn't find where this was introduced intentionally.

In v13.0.0: 
```yaml
name: Bindings
output: 'lib/generated_bindings.dart'
headers:
  entry-points:
    - 'src/temp.h'
  include-directives:
    - 'src/temp.h'
 ```
is enough to generate bindings. In v14.0.0, this generates a blank file. Instead, you need: 
```yaml
  include-directives:
    - '**/src/temp.h'
 ```

Seems like a pretty big bug to me, as it was causing all of my bindings to delete themselves upon re-generation. 

<details> 
<summary>Here is my <code>dart info</code></summary> 

```console
#### General info

- Dart 3.5.0 (stable) (Tue Jul 30 02:17:59 2024 -0700) on "windows_x64"
- on windows / "Windows 10 Home" 10.0 (Build 22631)
- locale is en-US
```

</details>

**Comment by dcharkes** [comment]
@liamappelbe could you ptal?

**Comment by liamappelbe** [comment]
@Levi-Lesches Does it work if you remove the `include-directives` option? Most of our integration tests don't use that option, so it's probably not as well covered as the rest.

**Comment by liamappelbe** [comment]
Seems to be related to the path that you invoke ffigen from. I couldn't repro it at all until I tried running ffigen from the same directory as the config file. It works fine if you go up a directory. Very odd.

**Comment by Levi-Lesches** [comment]
> Seems to be related to the path that you invoke ffigen from. I couldn't repro it at all until I tried running ffigen from the same directory as the config file. It works fine if you go up a directory. Very odd.

Yeah I figured as much when I realized adding the wildcards fixed it. Which isn't great because most people will be doing `dart run ffigen --config ffigen.yaml`, meaning, usually in the project root. Maybe getting it to output what it _thinks_ the relative path would help. 

>  Does it work if you remove the include-directives option? Most of our integration tests don't use that option, so it's probably not as well covered as the rest.

Yes, but then ffigen will generate everything from all header files which is usually not intended, especially when including standard header files.

**Comment by liamappelbe** [comment]
@Levi-Lesches The fix is published in v14.0.1

**Comment by Levi-Lesches** [comment]
Just tested it and it works, thanks!
